plugin/trino-phoenix5/src/test/java/io/trino/plugin/phoenix5/TestPhoenixConnectorTest.java

```diff
@@ -307,6 +317,69 @@ public void testMissingColumnsOnInsert()
         assertQuery("SELECT * FROM test_col_insert", "SELECT 1, 'val1', 'val2'");
     }
 
+    @Override
+    public void testTopNPushdown()
+    {
+        throw new SkipException("Phoenix does not support topN push down, but instead replaces partial topN with partial Limit.");
+    }
+
+    @Test
+    public void testReplacePartialTopNWithLimit()
+    {
+        assertThat(query("SELECT orderkey FROM orders ORDER BY orderkey LIMIT 10"))
+                .ordered()
+                .isNotFullyPushedDown(TopNNode.class, ExchangeNode.class, ExchangeNode.class, LimitNode.class);
+
+        assertThat(query("SELECT orderkey FROM orders ORDER BY orderkey DESC LIMIT 10"))
+                .ordered()
+                .isNotFullyPushedDown(TopNNode.class, ExchangeNode.class, ExchangeNode.class, LimitNode.class);
+
+        assertThat(query("SELECT * FROM orders ORDER BY orderkey, custkey LIMIT 10"))
+                .ordered()
+                .isNotFullyPushedDown(TopNNode.class, ExchangeNode.class, ExchangeNode.class, LimitNode.class);
+
+        assertThat(query("SELECT * FROM orders ORDER BY orderkey, custkey DESC LIMIT 10"))
+                .ordered()
+                .isNotFullyPushedDown(TopNNode.class, ExchangeNode.class, ExchangeNode.class, LimitNode.class);
+    }
+
+    @Test
+    public void testUseSortedProperties()
```

Comment:
It's unclear to me what this method is testing for? A comment or a better name may be helpful.
IIUC this is same as the previous test method but with multiple splits?

This test also looks different from the one in Hive. I'm not sure I understand why? Is it because in hive we need to tell the engine (via `propagate_table_scan_sorting_properties`) to trust that the table level sorting properties also apply to the partition level ones?

Reply:
The idea was to verify that the plan changes to use partial limit instead of topN and to verify basic sanity of query result when use of sorted table property is enabled through `propagate_table_scan_sorting_properties`.
In hive case, we explicitly fail queries if the partition level sorting turns out be incompatible with table level sorting. Also, someone might add an unsorted file to the table outside of trino, so we needed the escape hatch of `propagate_table_scan_sorting_properties`.

Reply:
Here the sorted property is coming through pushdown of partial topN into table scan rather than from some pre-defined table property, so maybe we can call this `testPartialLimitWithPreSortedInput` or something similar.

Reply:
But doesn't `isNotFullyPushedDown` cover both cases for Phoenix? It verifies plan shape and also verifies result correctness with pushdown disabled?

Reply:
I took the name from `TestHiveConnectorTest`. But agreed it's unclear.

Reply:
As for `assertPartialLimitWithPreSortedInputsCount` the only common place between Hive and Phoenix I can move this to is `BaseConnectorTest`. I'm going to do that and we can see how it looks :)

Reply:
Done. Let me know what you think.